grails-app/controllers/org/pih/warehouse/reporting/ConsumptionController.groovy

```diff
@@ -509,8 +512,21 @@ class ShowConsumptionCommand implements Validateable {
         parametersHash(nullable: true)
     }
 
-    def getAvailableProperties() {
-        Holders.grailsApplication.mappingContext.getPersistentEntity(Product.class.name).persistentPropertyNames
+    static String getTypePropertyName(Class type) {
```

Comment:
if we are going with this approach I think we need to add a comment above pointing out that these methods are copied from Grails 1 due to the fact that they are not present in Grails 3.

I would also suggest to move/create a utils class for this since this looks like a generic method that all of the `Command` classes are missing.

Reply:
I don't know if there is a sense to create a util class for that, because consumption report is the only place where it is used. Nowhere else this is needed.

Reply:
+1 for the util

Reply:
Moved to a util and made generic so that `getAvailableProperties` can accept a domain class.